cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,90 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import SignInPageObject from '../support/pages/signIn.pageObject';
+import SettingPageObject from '../support/pages/setting.pageObject';
+import homePageObject from '../support/pages/home.pageObject';
+
+const signInPage = new SignInPageObject(); 
+const settingPage = new SettingPageObject();
+const homePage = new homePageObject();
+
 describe('Settings page', () => {
+  let user;
+  let newUser;
   before(() => {
-
+    cy.task('db:clear');
   });
 
   beforeEach(() => {
-
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
+    });
+    cy.task('generateNewUser').then(generateNewUser => {
+        newUser = generateNewUser;
+    });
   });
 
   it('should provide an ability to update username', () => {
+    signInPage.visit();
+    cy.login(user.email, user.username, user.password);
+
+    settingPage.visit();
+    settingPage.usernameField
+      .clear()
+      .type(newUser.username)
 
+    settingPage.updateSettingsBtn
+      .click();
```

Comment:
you always click this button so move the click straight to the method

Reply:
Fixed